src/compiler/binder.ts

```diff
@@ -2300,7 +2303,21 @@ namespace ts {
             // When we create a property via 'exports.foo = bar', the 'exports.foo' property access
             // expression is the declaration
             setCommonJsModuleIndicator(node);
-            declareSymbol(file.symbol.exports, file.symbol, <PropertyAccessExpression>node.left, SymbolFlags.Property | SymbolFlags.ExportValue, SymbolFlags.None);
+            const lhs = node.left as PropertyAccessEntityNameExpression;
+            const symbol = forEachIdentifierInEntityName(lhs.expression, (id, original, e) => {
+                if (isExportsOrModuleExportsOrAlias(file, e) || (isIdentifier(e) && e.escapedText === "module" && original === undefined)) {

```

Comment:
I do not think we need to handle "module" in a special way. we probably should make forEachIdentifierInEntityName handle the terminal case correctly being `identifier` | `exports`  | `module.exports`, and return the correct symbol in each case.